defaults/main.yml

```diff
@@ -199,31 +202,39 @@ galaxy_app_config_default:
 
-# These are only used on Galaxy < 19.09
-galaxy_legacy_mutable_config:
-  shed_data_manager_config_file: "{{ galaxy_mutable_config_dir }}/shed_data_manager_conf.xml"
-  shed_tool_data_table_config: "{{ galaxy_mutable_config_dir }}/shed_tool_data_table_conf.xml"
-
-galaxy_legacy_config: "{{ galaxy_legacy_mutable_config if (galaxy_commit_id | replace('release_', '') | float() < 19.09) else {} }}"
-  
 # Need to set galaxy_config_default[galaxy_app_config_section] dynamically but Ansible/Jinja2 does not make this easy
 galaxy_config_default: "{{ {} | combine({galaxy_app_config_section: galaxy_app_config_default}) }}"
-galaxy_config_with_legacy: "{{ galaxy_config_default | combine(galaxy_legacy_config | default({}), recursive=True) }} "
-galaxy_config_merged: "{{ galaxy_config_with_legacy | combine(galaxy_config | default({}), recursive=True) }}"
+galaxy_config_merged: "{{ galaxy_config_default | combine(galaxy_config | default({}), recursive=True) }}"
 
 # Automatically instantiate mutable config files if they don't exist (dest will not be overwritten if it exists)
-galaxy_mutable_configs:
+galaxy_legacy_mutable_configs:
   - src: "{{ galaxy_server_dir }}/config/migrated_tools_conf.xml.sample"
     dest: "{{ galaxy_config_merged[galaxy_app_config_section].migrated_tools_config }}"
   - src: "{{ galaxy_server_dir }}/config/shed_data_manager_conf.xml.sample"
     dest: "{{ galaxy_config_merged[galaxy_app_config_section].shed_data_manager_config_file }}"
   - src: "{{ galaxy_server_dir }}/config/shed_tool_data_table_conf.xml.sample"
     dest: "{{ galaxy_config_merged[galaxy_app_config_section].shed_tool_data_table_config }}"
 
-galaxy_mutable_config_templates:
+galaxy_base_mutable_config_templates:
   - src: "shed_tool_conf.xml"
     dest: "{{ galaxy_shed_tool_conf_file }}"
 
+# Automatically instantiate mutable config files from copies in role if they don't exist (for Galaxy 19.09 and later)
+galaxy_mutable_configs:
+  - src: "{{ role_path }}/files/shed_data_manager_conf.xml"
+    dest: "{{ galaxy_config_merged[galaxy_app_config_section].shed_data_manager_config_file }}"
+  - src: "{{ role_path }}/files/shed_tool_data_table_conf.xml"
+    dest: "{{ galaxy_config_merged[galaxy_app_config_section].shed_tool_data_table_config }}"
+    
+# required for Galaxy release 19.09 and later
+galaxy_extra_mutable_config_templates:
+  - src: "migrated_tools_conf.xml.j2"
+    dest: "{{ galaxy_config_merged[galaxy_app_config_section].migrated_tools_config }}"
+
+# produce the galaxy_mutable_config_templates differently depending on Galaxy release
+# older versions used the value from galaxy_base_mutable_config_templates, newer versions combine that with galaxy_extra_mutable_config_templates
+galaxy_mutable_config_templates: "{{ (galaxy_base_mutable_config_templates + galaxy_extra_mutable_config_templates) if (galaxy_commit_id | replace('release_', '') | float() >= 19.09) else galaxy_base_mutable_config_templates | default([]) }}"    
```

Comment:
This isn't guaranteed to work since the commit ID is not necessarily a release branch (or a branch at all). We have a task-based method for getting the major version in [tasks/_inc_galaxy_version.yml](https://github.com/galaxyproject/ansible-galaxy/blob/master/tasks/_inc_galaxy_version.yml). If you include this before you need `galaxy_mutable_config_templates` (but after clone/update tasks) you can use `__galaxy_major_version is version('19.09', '>=')` here.